src/main/cpp/tests/cast_string.cpp

```diff
@@ -229,4 +231,212 @@ TYPED_TEST(StringToIntegerTests, Overflow)
   }();
 
   CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
-}
\ No newline at end of file
+}
+
+TEST_F(StringToDecimalTests, Simple)
+{
+  auto const strings = test::strings_column_wrapper({"1", "0", "-1"});
+  strings_column_view scv{strings};
+
+  auto const result =
+    spark_rapids_jni::string_to_decimal(1, 0, scv, false, rmm::cuda_stream_default);
+
+  test::fixed_point_column_wrapper<int32_t> expected({1, 0, -1}, {1, 1, 1}, numeric::scale_type{0});
+
+  test::print(result->view());
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUAL(result->view(), expected);
+}
+
+TEST_F(StringToDecimalTests, OverPrecise)
+{
+  auto const strings = test::strings_column_wrapper({"123456", "999999", "-123456", "-999999"});
+  strings_column_view scv{strings};
+
+  auto const result =
+    spark_rapids_jni::string_to_decimal(5, 0, scv, false, rmm::cuda_stream_default);
+
+  test::fixed_point_column_wrapper<int32_t> expected(
+    {12346, 100000, -12346, -100000}, {0, 0, 0, 0}, numeric::scale_type{0});
```

Comment:
nit: This test is a little confusing. With a precision of 5 and a scale of 0 all of the numbers overflow and you get null back.  All of that is great. The confusing part is that you don't have an XXX like in other code that indicates it is a NULL placeholder so I don't spend a few mins trying to understand why these values look like they are rounded almost correctly.

Reply:
I agree and this is a result of copy/paste. I have changed this to 0's, but would you prefer a null variable name instead?

Reply:
That is good enough for me.